### ISSUE TITLE
Add support support for qualified calls

### ISSUE THREAD

**Issue by momvart** [Kobzol/rust-delegate#52]
Consider that the target implements two traits that have a method with the same name. In this case, barely calling the function over the target is ambiguous for the compiler. For example:
```rust
trait A {
    fn f(&self);
}

trait B {
    fn f(&self);
}

struct Foo;

impl A for Foo { /* ... */ }
impl B for Foo { /* ... */ }

struct Bar(Foo);

impl A for Bar {
    delegate! {
        to &self.0 {
            fn f(&self);  // Error: multiple applicable items in scope
        }
    }
}
```
This can be fixed by generating a body that calls the function in a qualified way, i.e., `A::f(&self.0);`. However, I guess the trait name is not available to the macro so it cannot do that by default.

A solution can be an attribute that takes the trait's name.
```rust
delegate! {
    to &self.0 {
        #[through(A)]
        fn f(&self);
    }
}
```

**Comment by Kobzol** [comment]
Sorry for the delay, let me know if [this](https://github.com/Kobzol/rust-delegate/pull/54) solves your use-case.

You can try it like this:
```toml
[dependencies]
delegate = { git = "https://github.com/kobzol/rust-delegate", branch = "attr-through" }
```

**Comment by momvart** [comment]
Awesome.
I would like to add the following test case which tests overriding the segment's attribute with the method's, and looks more realistic at least based on my case.
```rust
#[test]
fn test_override_through_trait() {
    use delegate::delegate;

    trait A {
        fn f(&self) -> u32;
        fn g(&self) -> u32;
    }

    trait B {
        fn g(&self) -> u32;
    }

    struct Foo;

    impl A for Foo {
        fn f(&self) -> u32 {
            0
        }

        fn g(&self) -> u32 {
            0
        }
    }
    impl B for Foo {
        fn g(&self) -> u32 {
            1
        }
    }

    struct Bar(Foo);

    impl Bar {
        delegate! {
            #[through(A)]
            to &self.0 {
                fn f(&self) -> u32;
                #[through(B)]
                fn g(&self) -> u32;
            }
        }
    }

    let bar = Bar(Foo);
    assert_eq!(bar.f(), 0);
    assert_eq!(bar.g(), 1);
}

```

**Comment by Kobzol** [comment]
Good idea, I added this to an [existing test](https://github.com/Kobzol/rust-delegate/pull/54/files#diff-f0664a4d2dfd09e81133ca99513285e5e0683b08a94e2fecc3a9c655d84d374cR179).

**Comment by Kobzol** [comment]
Just to be clear: did the PR work for you, can I merge it and consider this issue to be resolved?

**Comment by Kobzol** [comment]
Merged and released version `0.10.0`.
